src/pip/_internal/vcs/versioncontrol.py

```diff
@@ -71,6 +81,104 @@ def make_vcs_requirement_url(repo_url, rev, project_name, subdir=None):
     return req
 
 
+def call_subprocess(
+    cmd,  # type: Union[List[str], CommandArgs]
+    cwd=None,  # type: Optional[str]
+    on_returncode='raise',  # type: str
```

Comment:
I would suggest to always raise in this function, and catch at the call sites instead, like how `subprocess.check_output()` on Python 3. This will make it easier to simplify the implementation once we drop Python 2 as well.

Reply:
Okay, in that case I can drop the argument from this call, and the corresponding code which handles conditions for `on_returncode` apart from raise.

Does that means I can drop `extra_ok_returncodes` from this function too? Since we are always going to raise and `extra_ok_returncodes` which don't raise errors for specified return codes won't be used.

Also what do we want to do with `log_failed_cmd` and `extra_environ` parameter (which is only being used [here](https://github.com/pypa/pip/blob/master/tests/functional/test_vcs_git.py#L65) and [here](https://github.com/pypa/pip/blob/master/tests/functional/test_vcs_git.py#L82).